_posts/2019-08-09-iglu-scala-client-0.6.0-released/index.md

```diff
@@ -0,0 +1,116 @@
+* [Cats][cats] is used instead of [Scalaz][scalaz] as FP library of choice. Cats is a transitive dependency of Circe and Cats Effect, which makes it a natural choice, since no other dependencies are shipped with Scalaz.
+* [networknt json-schema-validator][networknt-validator] is used instead of FGE JSON Validator. One more change library, driving our bad rows effort.  networknt json-schema-validator is actively maintained, provides clean API and shows very impressive results in benchmarks
+
+You can find more usage examples on dedicated [wiki page][iglu-client-docs].
+
+<h3 id="semantic-changes">2. Semantic Changes</h3>
+
+In a batch ETL world, we tried to reduce the load on Iglu Registries by leveraging very simple retry-and-cache algorithm, that was making some configurable amount of attempts before deciding that schema is missing or invalid and caching this failure. The only thing that potentially could reset this cached value is cacheTtl property, that would force the resolver to retry no matter whether cached value is success (in case somebody mutated schema) or failure (in case the registry had a long outage).
+
+This approach does not work for RT-first world anymore. There's no meaningful amount of attempts that resolver needs to make before considering a schema missing or invalid. Streaming application can keep working for many weeks without restart and if during this time, one registry goes down for couple of minutes and resolver will try to resolve a schema it means that until next TTL eviction all data will be invalid. And retries won't help here because they all will happen in a short period of time. 
+
+However, we still need to have certain retry behavior, because registries always can go offline. In a streaming world, the best practice for retries is backoff period. In 0.6.0 the Iglu resolver will attempt to refetch failed schemas with steadily growing period of time between attempts. This period grows from subsecond delays to approximately 20 minutes. What is also very important, these re-attempts will be made only for non-successful responses.
+
+ResolutionError (subtype of `ClientError`) has two properties to reflect the history of attempts: lastAttempt a timestamp of last attempt being made and attempts showing amount of attempts taken so far.
+
+
+<h3 id="new-validator">3. New Validator</h3>
+
+As it was mentioned before, Iglu Scala Client uses new [JSON Schema validator][networknt-validator] under the hood (which hover can be replaced with any custom one). Despite this validator also targets JSON Schema spec v4, it nevertheless can have incompatibilities with our previous JSON Schema validator. As a result some instances that were considered valid by Iglu Scala Client pre-0.6.0 can now be silently invalidated.
+
+Here's the short list of most widely used Snowplow components we're planning to release with Iglu Client 0.6.0:
```

Comment:
Here's a short list of the most widely used Snowplow components we're planning to release with Iglu Client 0.6.0: